airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/utils/compare.py

```diff
@@ -52,13 +51,20 @@ def diff_dicts(left, right, use_markup) -> Optional[List[str]]:
 
 
 @functools.total_ordering
-class DictWithHash(dict):
+class HashMixin:
+    _hash = None
```

Comment:
We shouldn't be persisting the hash value in the object because this Mixin is designed to be used with mutable objects (dict and list). e.g.
```
a = DictWithHashMixin( {"one": 1} )
b = DictWithHashMixin( {"one": 1} )
a == b  # this will correctly be True
a["two"] = 2
a == b  # this will now INCORRECTLY be True, because we're comparing the same saved hash as before
# meaning hash(a) == hash(b) when it should be different

Reply:
Based on current usage, yes... but ultimately this PR implements a new Dict and List that don't update their hash after being mutated. For obvious reasons that behaviour would not be assumed by a developer in the future utilising these and we could end up with problems. I think we should therefore always recompute.

I would be very surprised if performance from Python's hash function became limiting.

Reply:
this make sense, updated PR